pkg/odo/genericclioptions/runnable.go

```diff
@@ -80,16 +81,36 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
 	var err error
 	startTime := time.Now()
 	cfg, _ := preference.NewClient()
-	disableTelemetry, _ := strconv.ParseBool(os.Getenv(segment.DisableTelemetryEnv))
+	//lint:ignore SA1019 We deprecated this env var, but until it is removed, we still need to support it
+	disableTelemetryValue, disableTelemetryEnvSet := os.LookupEnv(segment.DisableTelemetryEnv)
+	disableTelemetry, _ := strconv.ParseBool(disableTelemetryValue)
 	debugTelemetry := segment.GetDebugTelemetryFile()
+	trackingConsent := os.Getenv(segment.TrackingConsentEnv)
+
+	// check for conflicting settings
+	if disableTelemetryEnvSet && ((disableTelemetry && trackingConsent == "yes") || (!disableTelemetry && trackingConsent == "no")) {
+		//lint:ignore SA1019 We deprecated this env var, but we really want users to know there is a conflict here
+		util.LogErrorAndExit(
+			fmt.Errorf("%[1]s and %[2]s values are in conflict. %[1]s is deprecated, please use only %[2]s",
+				segment.DisableTelemetryEnv, segment.TrackingConsentEnv), "")
+	}
 
 	// Prompt the user to consent for telemetry if a value is not set already
 	// Skip prompting if the preference command is called
 	// This prompt has been placed here so that it does not prompt the user when they call --help
 	if !cfg.IsSet(preference.ConsentTelemetrySetting) && cmd.Parent().Name() != "preference" {
 		if !segment.RunningInTerminal() {
 			klog.V(4).Infof("Skipping telemetry question because there is no terminal (tty)\n")
+		} else if trackingConsent == "no" {
+			klog.V(4).Infof("Skipping telemetry question due to %s=%s\n", segment.TrackingConsentEnv, trackingConsent)
+		} else if trackingConsent == "yes" {
+			klog.V(4).Infof("Skipping telemetry question due to %s=%s\n", segment.TrackingConsentEnv, trackingConsent)
+			klog.V(4).Info("Telemetry is enabled!\n")
+			if err1 := cfg.SetConfiguration(preference.ConsentTelemetrySetting, "true"); err1 != nil {
+				klog.V(4).Info(err1.Error())
+			}
 		} else if disableTelemetry {
+			//lint:ignore SA1019 We deprecated this env var, but until it is removed, we still need to support it
 			klog.V(4).Infof("Skipping telemetry question due to %s=%t\n", segment.DisableTelemetryEnv, disableTelemetry)
 		} else {
 			var consentTelemetry bool
```

Comment:
You can also use isTrackingConsentEnabled in this block

Reply:
Good catch - thanks. Done in [`a4415ec` (#6258)](https://github.com/redhat-developer/odo/pull/6258/commits/a4415eca1e0856b8486ddb4e4d75934b92876046)

Reply:
Arf, sorry - I changed in [`8b69040` (#6258)](https://github.com/redhat-developer/odo/pull/6258/commits/8b69040e10a5d5f743460946a43bbc72f7181e1f). I checked and it should be ok now.